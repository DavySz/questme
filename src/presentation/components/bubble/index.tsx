import { BubbleFill } from "./fill";
import { BubbleRoot } from "./root";
import { BubbleOutline } from "./outline";

export const Bubble = {
  Root: BubbleRoot,
  Fill: BubbleFill,
  Outline: BubbleOutline,
};
