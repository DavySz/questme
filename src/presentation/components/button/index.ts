import { ButtonIcon } from "./icon";
import { ButtonRoot } from "./root";
import { ButtonText } from "./text";

export const Button = {
  Root: ButtonRoot,
  Text: ButtonText,
  Icon: ButtonIcon,
};
