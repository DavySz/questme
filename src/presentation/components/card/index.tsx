import { CardGroup } from "./group";
import { CardIcon } from "./icon";
import { CardRoot } from "./root";

export const Card = {
  Root: CardRoot,
  Icon: CardIcon,
  Group: CardGroup,
};
