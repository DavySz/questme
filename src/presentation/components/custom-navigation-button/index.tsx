import { CustomNavigationIcon } from "./icon";
import { CustomNavigationButtonRoot } from "./root";

export const CustomNavigationButton = {
  Root: CustomNavigationButtonRoot,
  Icon: CustomNavigationIcon,
};
