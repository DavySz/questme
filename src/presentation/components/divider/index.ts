import { DividerRoot } from "./root";
import { DividerText } from "./text";

export const Divider = {
  Root: DividerRoot,
  Text: DividerText,
};
