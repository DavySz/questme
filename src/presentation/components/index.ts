export * from "./button";
export * from "./text";
export * from "./screen-template";
export * from "./divider";
