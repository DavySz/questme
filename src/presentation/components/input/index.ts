import { InputIcon } from "./icon";
import { InputLabel } from "./label";
import { InputRoot } from "./root";
import { InputText } from "./text";

export const Input = {
  Root: InputRoot,
  Label: InputLabel,
  Text: InputText,
  Icon: InputIcon,
};
