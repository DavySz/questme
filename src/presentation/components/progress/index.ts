import { ProgressBar } from "./bar";
import { ProgressLabel } from "./label";
import { ProgressRoot } from "./root";

export const Progress = {
  Root: ProgressRoot,
  Label: ProgressLabel,
  Bar: ProgressBar,
};
