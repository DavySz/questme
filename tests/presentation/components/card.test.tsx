import { Card } from "@presentation/components";
import { customRender } from "../../utils";

import PlusIcon from "@presentation/assets/icons/plus-icon.svg";
import { Text } from "react-native";

describe("Card", () => {
  test("Should render component without errors", () => {
    const tree = customRender(
      <Card.Root>
        <Card.Icon icon={PlusIcon} />
        <Card.Group>
          <Text>any-children</Text>
        </Card.Group>
      </Card.Root>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("Should render correct icon when color is passed", () => {
    const { findByTestId } = customRender(
      <Card.Root>
        <Card.Icon icon={PlusIcon} color="neutral-black" />
      </Card.Root>
    );

    expect(findByTestId("icon-with-color")).toBeTruthy();
  });
});
