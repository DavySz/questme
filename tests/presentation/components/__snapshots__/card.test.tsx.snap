// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should render component without errors 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderColor": "#EFEEFC",
      "borderStyle": "solid",
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
      "width": "100%",
    }
  }
>
  <SvgMock
    testID="icon-without-color"
  />
  <View
    style={
      {
        "gap": 6,
      }
    }
  >
    <Text>
      any-children
    </Text>
  </View>
</View>
`;
